[Questions] Error: exit with Code 2 when running github actions build

### Have you read the FAQ?

- [X] Yes, I did read the FAQ

### Is there an existing issue/question for this?

- [X] I have searched the existing issues

### Do you think this is a bug?

❌ No, I don't think this is a bug. I will explain below

### My question

Im trying to build using github actions and I have a decrypted youtube IPA with a link that seems to be working but I keep getting the following error. 
<img width="1230" alt="image" src="https://github.com/qnblackcat/uYouPlus/assets/77735965/46b3fde1-bbaa-41e9-b82a-6e445c70b136">
This is when its trying to Fix Compiling and Build Package.
the error is :
clang: error: linker command failed with exit code 1 (use -v to see invocation)
and then
Error: Process completed with exit code 2.

Does anyone know why this might be happening?


### Additional context

_No response_

whats a bit strange is that before we had the error too but it didn't fail:

This worked
```
Run cd /Users/runner/work/uYouPlus/uYouPlus/main
==> Downloading uYou…
> Making all in Tweaks/Alderis…
> Making all for xcodeproj Alderis…
    Run script build phase 'SwiftLint' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Alderis' from project 'Alderis')
▸ Copying AlderisSDKCompatibility.h
▸ Copying Alderis.h
▸ Processing Info.plist
▸ Compiling Alderis_vers.c
▸ Linking Alderis
▸ Generating 'Alderis.framework.dSYM'
▸ Running script 'SwiftLint'
▸ Touching Alderis.framework (in target 'Alderis' from project 'Alderis')
▸ Archive Succeeded
==> Signing Alderis…
> Making all in lcpshim…
> Making all for library libcolorpicker…
==> Compiling PFColorCell.m (arm64)…
==> Compiling PFColorAlert.m (arm64)…
==> Compiling ColorFunctions.m (arm64)…
==> Compiling PFSimpleLiteColorCell.m (arm64)…
==> Compiling HBColorPickerTableCell.m (arm64)…
==> Compiling UIColor+PFColor.m (arm64)…
==> Compiling PFLiteColorCell.m (arm64)…
ld: warning: directory not found for option '-F/Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/install/Library/Frameworks'
==> Linking library libcolorpicker (arm64)…
ld: warning: directory not found for option '-F/Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/install/var/jb/Library/Frameworks'
==> Generating debug symbols for libcolorpicker…
==> Stripping libcolorpicker (arm64)…
==> Merging library libcolorpicker…
==> Signing libcolorpicker…
> Making all in Tweaks/FLEXing/libflex…
> Making all for tweak libFLEX…
==> Preprocessing libFLEX.x…
==> Compiling FLEX/Classes/Utility/Runtime/flex_fishhook.c (arm64)…
```

This fails
```
Fix compiling & Build Package
==> Downloading uYou…
> Making all in Tweaks/Alderis…
> Making all for xcodeproj Alderis…
▸ Copying AlderisSDKCompatibility.h
▸ Copying Alderis.h
▸ Processing Info.plist
▸ Compiling Alderis_vers.c
▸ Linking Alderis
▸ Generating 'Alderis.framework.dSYM'
▸ Running script 'SwiftLint'
    SwiftLint not installed, download from https://github.com/realm/SwiftLint
▸ Touching Alderis.framework (in target 'Alderis' from project 'Alderis')
    Run script build phase 'SwiftLint' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Alderis' from project 'Alderis')
▸ Archive Succeeded
==> Signing Alderis…
> Making all in lcpshim…
> Making all for library libcolorpicker…
==> Compiling ColorFunctions.m (arm64)…
==> Compiling PFColorCell.m (arm64)…
==> Compiling PFColorAlert.m (arm64)…
==> Compiling PFLiteColorCell.m (arm64)…
==> Compiling PFSimpleLiteColorCell.m (arm64)…
==> Compiling UIColor+PFColor.m (arm64)…
==> Compiling HBColorPickerTableCell.m (arm64)…
==> Linking library libcolorpicker (arm64)…
ld: warning: -multiply_defined is obsolete
ld: warning: ignoring duplicate libraries: '-lc++'
ld: warning: search path '/Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/install/Library/Frameworks' not found
ld: warning: search path '/Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/install/var/jb/Library/Frameworks' not found
ld: framework 'Preferences' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[5]: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/instance/library.mk:52: /Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/arm64/libcolorpicker.dylib] Error 1
make[4]: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/instance/library.mk:52: /Users/runner/work/uYouPlus/uYouPlus/main/.theos/obj/arm64/libcolorpicker.dylib] Error 2
make[3]: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/instance/library.mk:[37](https://github.com/m1337v/uYouPlus/actions/runs/8815794708/job/24198483563#step:11:38): internal-library-all_] Error 2
make[2]: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/master/rules.mk:146: libcolorpicker.all.library.variables] Error 2
make[1]: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/master/aggregate.mk:12: internal-all] Error 2
make: *** [/Users/runner/work/uYouPlus/uYouPlus/theos/makefiles/master/aggregate.mk:12: internal-all] Error 2
Error: Process completed with exit code 2.
```

--- a/.github/workflows/buildapp.yml
+++ b/.github/workflows/buildapp.yml
@@ -44,7 +44,7 @@ concurrency:
 jobs:
   build:
     name: Build uYouPlus
-    runs-on: macos-latest
+    runs-on: macos-12
     permissions:
       contents: write
 

